python/lsst/sphgeom/pixelizations.i

```diff
@@ -41,8 +41,8 @@
             o = PyNumber_Multiply(tmp, tmp);
             Py_DECREF(tmp);
         } else {
-            if (p <= static_cast<unsigned long long>(PyInt_GetMax())) {
-                o = PyInt_FromLong(static_cast<long>(p));
+            if (p <= static_cast<unsigned long long>(std::numeric_limits<long>::max())) {
+                o = PyLong_FromLong(static_cast<long>(p));
```

Comment:
Actually, if we avoid `PyInt_FromLong` even on Python 2, you can simplify this function to:

    inline PyObject * _rangeBeginningOrEndToIntOrLong(uint64_t p, bool isEndPoint) {
        PyObject * o;
        if (p == 0 && isEndPoint) {
            // Map trailing bookend to 2**64
            PyObject * tmp = PyLong_FromUnsignedLongLong(UINT64_C(0x100000000));
            if (!tmp) {
                return nullptr;
            }
            o = PyNumber_Multiply(tmp, tmp);
            Py_DECREF(tmp);
        } else {
            o = PyLong_FromUnsignedLongLong(p);
        }
        return o;
    }